x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/FilterBlockBuilder.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.data;
+
+public class FilterBlockBuilder {
+    private Block block;
+    private int[] positions;
+
+    public FilterBlockBuilder withBlock(Block block) {
+        this.block = block;
+        return this;
+    }
+
+    public FilterBlockBuilder withPositions(int... positions) {
+        this.positions = positions;
+        return this;
+    }
+
+    public Block build() {
+        BlockFactory blockFactory = block.blockFactory();
+        long ramBytesUsed = block.ramBytesUsed();
+
+        Block filteredBlock = block.filter(positions);
+        block.close();
+        blockFactory.adjustBreaker(ramBytesUsed, true);
```

Comment:
I'm not sure I see the value here, rather than just updating the filter implementations of each block to do this?

Reply:
Works for me as well and was how I'd initially wanted to do it - I think I misunderstood your comment that we'd want to keep block construction and memory accounting separate.